apps/server/src/modules/h5p-editor/service/h5p-translator.service.ts

```diff
@@ -0,0 +1,34 @@
+import { ITranslationFunction } from '@lumieducation/h5p-server';
+import i18next from 'i18next';
+import i18nextFsBackend from 'i18next-fs-backend';
+import path from 'path';
+import { translatorConfig } from '../h5p-editor.config';
+
+export const Translator = {
+	async translate() {
+		const lumiPackagePath = path.dirname(require.resolve('@lumieducation/h5p-server/package.json'));
+		const pathBackend = path.join(lumiPackagePath, 'build/assets/translations/{{ns}}/{{lng}}.json');
+
+		const translationFunction = await i18next.use(i18nextFsBackend).init({
+			backend: {
+				loadPath: pathBackend,
+			},
+			ns: [
+				'client',
+				'copyright-semantics',
+				'hub',
+				'library-metadata',
+				'metadata-semantics',
+				'mongo-s3-content-storage',
+				's3-temporary-storage',
+				'server',
+				'storage-file-implementations',
+			],
+			preload: translatorConfig.AVAILABLE_LANGUAGES,
+		});
+
+		const translate: ITranslationFunction = (key, language) => translationFunction(key, { lng: language });
+
+		return translate;
+	},
+};
```

Comment:
```suggestion
};

export const translate = await Translator.translate();
```
this is for discusion with martin

Reply:
still open @SteKrause , dont wana change it?

Reply:
This solution throws a warning with the suggestion:
"Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher."

@MartinSchuhmacher should I change the module options accordingly to the warning?
Or can the code remain as is?

Reply:
Changes resolved